src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1710,6 +1710,55 @@ public async Task<bool> IsAsyncMethod(SessionId sessionId, int methodId, Cancell
             return retDebuggerCmdReader.ReadByte() == 1 ; //token
         }
 
+        public async Task<JArray> GetHoistedLocalVariables(SessionId sessionId, int objectId, JArray asyncLocals, CancellationToken token)
+        {
+            JArray asyncLocalsFull = new JArray();
+            List<int> objectsAlreadyRead = new();
+            objectsAlreadyRead.Add(objectId);
+            foreach (var asyncLocal in asyncLocals)
+            {
+                var fieldName = asyncLocal["name"].Value<string>();
+                Console.WriteLine($"olha aqui thays {fieldName}");
+                if (fieldName.EndsWith("__this"))
+                {
+                    asyncLocal["name"] = "this";
+                    asyncLocalsFull.Add(asyncLocal);
+                }
+                else if (fieldName.StartsWith("<>8__") ||
+                        fieldName.StartsWith("<>f__ref") ||
+                        fieldName.StartsWith("$locvar") ||
+                        fieldName.StartsWith("CS$<>")) //same code that has on debugger-libs
+                {
+                    if (DotnetObjectId.TryParse(asyncLocal?["value"]?["objectId"]?.Value<string>(), out DotnetObjectId dotnetObjectId))
+                    {
+                        if (int.TryParse(dotnetObjectId.Value, out int objectIdToGetInfo))
+                        {
+                            if (!objectsAlreadyRead.Contains(objectIdToGetInfo))
+                            {
+                                var asyncLocalsFromObject = await GetObjectValues(sessionId, objectIdToGetInfo, true, false, false, false, token);
+                                var hoistedLocalVariable = await GetHoistedLocalVariables(sessionId, objectIdToGetInfo, asyncLocalsFromObject, token);
+                                asyncLocalsFull = new JArray(asyncLocalsFull.Union(hoistedLocalVariable));
+                            }
+                        }
+                    }
+                }
+                else if (fieldName.StartsWith("<>")) //examples: <>t__builder, <>1__state
+                {
+                    continue;
+                }
+                else if (fieldName.StartsWith('<')) //examples: <code>5__2
+                {
+                    asyncLocal["name"] = Regex.Match(fieldName, @"\<([^)]*)\>").Groups[1].Value;
```

Comment:
It might make sense to cache the compiled regex

Reply:
Done.